tmt/libraries/beakerlib.py

```diff
@@ -255,11 +256,27 @@ def fetch(self) -> None:
                     # Use the default branch if no ref provided
                     if self.ref is None:
                         self.ref = self.default_branch
+                    try:
+                        plan = cast(Discover, self.parent.parent).plan
```

Comment:
This could be quite a bold assignment and it seems it is causing failures in unit tests like
https://github.com/teemtee/tmt/blob/main/tests/unit/test_beakerlib.py#L13

I do see some options here. 
 - Add extra check so this assignment is attempted only when `ref` contain dynamic reference and in such a case it would raise an exception when plan cannot be reached.
 - Modify unit tests so that they have a plan
 - Modify `resolve_dynamic_ref()` so it accepts `None` for the `plan` parameter and raise exception` later in the code when plan/context is needed but not available.
I think I will go with option 3 for now.

Reply:
Option number 3 sounds good, as `plan` is not needed for all actions taken by `resolve_dynamic_ref()`.